src/floor/RequestSimulator.java

```diff
@@ -24,35 +27,76 @@ public RequestSimulator(int f) {
 		}
 
 		floorNum = f;
+		time = 0;
 	}
 
 	@Override
 	public void run() {
+		ElevatorRequestMessage m;
+		FloorMetaMessage f;
+		
+		try (BufferedReader br = new BufferedReader(new FileReader(SimulationVars.inputFile))) {
+			String line;
+			while ((line = br.readLine()) != null) {
+				String[] parts = line.split(" ");
+				String timeOfReq = parts[0];
+				String originFloor = parts[1];
+				String direction = parts[2];
+				String destination = parts[3];
 
-		try {
-			Thread.sleep(2000);
-		} catch (InterruptedException e) {
+				//our next message
+				if (Integer.parseInt(originFloor) == floorNum) {
+					//make the messages
+					m = new ElevatorRequestMessage();
+					if (direction.equals("Up")) {
+						m.setDirection(Direction.UP);
+					} else {
+						m.setDirection(Direction.DOWN);
+					}
+					m.setOriginFloor(floorNum);
+					byte[] schedulerData = Message.serialize(m);
+					DatagramPacket schedulerPacket = new DatagramPacket(schedulerData, schedulerData.length, SimulationVars.schedulerAddress, SimulationVars.schedulerPort);
+					
+					f = new FloorMetaMessage(true);
+					f.setStartingFloor(floorNum);
+					f.setDestinationFloor(Integer.parseInt(destination));
+					byte[] metaData = Message.serialize(f);
+					DatagramPacket metaPacket = new DatagramPacket(metaData, metaData.length, SimulationVars.floorAddresses[floorNum], SimulationVars.floorPorts[floorNum]);
+					
+					//sleep until its time to send
+					try {
+						Thread.sleep(getSleepTime(timeOfReq));
```

Comment:
U shouldn't sleep while reading a buffer. Free the file as soon as possible.

Reply:
In my testing I saw that multiple threads can have their noses in the same file out once so I didn't see this as an issue but I like the idea.